### PR TITLE
Change thread default to INT_MAX

### PR DESCRIPTION
**Pull request by ncteisen**
To fix internal failure

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]      +8  +0.0%

  [ = ]       0 TOTAL       +8  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/cpp/server/server_cc.cc [hunk 1]
```diff
@@ -48,10 +48,11 @@ namespace grpc {
 namespace {
 
 // The default value for maximum number of threads that can be created in the
-// sync server. This value of 500 is empirically chosen. To increase the max
-// number of threads in a sync server, pass a custom ResourceQuota object (with
-// the desired number of max-threads set) to the server builder
-#define DEFAULT_MAX_SYNC_SERVER_THREADS 500
+// sync server. This value of INT_MAX is chose to match the default behavior if
```
Comment (yashykt) [inline src/cpp/server/server_cc.cc]: /s/chose/chosen

Comment (ncteisen) [reply src/cpp/server/server_cc.cc]: Done

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]      +8  +0.0%

  [ = ]       0 TOTAL       +8  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,952,734      Total (=)      1,952,734

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,663,573      Total (<)     10,663,580

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,952,734      Total (=)      1,952,734

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,663,577      Total (<)     10,663,580

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by vjpai** [comment]
I have confirmed at my desktop that the thread-constrained version of thread_stress_test in this configuration sees 99.8% resource exhausted RPCs but survives anyway. So, we are indeed returning resource_exhausted as expected.

**Pull request merged**